File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)=='(')
                stack.push('(');
            else{
                if(!stack.isEmpty() && stack.peek()=='(')
                    stack.pop();
                else if(!stack.isEmpty() && stack.peek()==')')
                    return false;
                else if(stack.isEmpty()) return false;
            }
        }
        if(stack.isEmpty()) return true;
        return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<int> st;
    for(char c : s){
        if(c=='(') st.push('(');
        else {
            if(st.empty()) return false;
            else st.pop();
        }
    }

    if(st.empty()) return true;
    return false;
}